dbms/src/Interpreters/evaluateConstantExpression.cpp

```diff
@@ -102,8 +121,24 @@ namespace
             if (name == identifier->name)
             {
                 ColumnWithTypeAndName column;
+                Field value;
+
+                try
+                {
+                    value = convertFieldToType(literal->value, *type);
+                }
+                catch (const Exception &e)
```

Comment:
This is very hackish, and can be replaced with some visitor that will verify that we have only const type eventually?